awx/main/models/events.py

```diff
@@ -325,6 +327,11 @@ def _update_from_event_data(self):
         if isinstance(res, dict):
             if res.get('changed', False):
                 self.changed = True
+        if isinstance(self, JobEvent) and not self.job.gather_event_types == 'all': # Event filtering state machine
+            if not self.failed and self.event_level in [0,3] and self.job.gather_event_types == "errors":
```

Comment:
My brain found this code kind of hard to parse.  What about something like:

```python
if self.job.gather_event_types in ("none", "errors"):
    self.skip_save = self.event_level in [0,3]  # skip runner_on and meta events (like verbose/debug/warnings)
    if self.job.gather_event_types == "errors":
        self.skip_save = self.skip_save and not self.failed  # skip failures/errors
```

Reply:
Also, *should* we skip level zero entirely?

<img width="690" alt="image" src="https://user-images.githubusercontent.com/214912/78724279-1bfe5a80-78fb-11ea-99b3-7adfc15b3617.png">

It seems to me like _most_ of these are actually the types of things you'd maybe want to record.  Maybe we should just exclude "verbose" events?

Reply:
debug too perhaps? deprecated and warning are also typically filtered out by folks?

Reply:
it *will* pick up true errors as identified here.